### PR TITLE
Fixes for compiling with Visual Studio 2013.  This addresses

### PR DESCRIPTION
**Pull request by cffk**
(1) lack of support for constexpr
(2) absence of {} as an initializer
(3) need to have consistent return types

Fixing these doesn't detract too much from the expressive power of
C++11.  I recommend adding Visual Studio 2013 and Visusal Studio 2015
compilers to the continuous integration suite!

### CODE REVIEW

**Comment by hobu** [comment]
> I recommend adding Visual Studio 2013 and Visusal Studio 2015 compilers to the continuous integration suite

Feel free to make a branch and start modifying appveyor.yml to get both permutations building.

**Comment by gadomski** [comment]
@cffk I've got a branch to fix that AppVeyor file utils break at https://github.com/gadomski/PDAL/tree/glob-test that I'm just cleaning up before submitting. If you want to pick up that test fix, just base your changes off of my branch once I open a pull request (I'll drop the PR number in this ticket).

**Comment by gadomski** [comment]
@cffk #1362 is the PR w/ the (hopeful) AppVeyor fix.

**Comment by cffk** [comment]
Hmm, the build failure looks like some random Windows lossage.  Is there some
easy way to trigger another attempt?

**Comment by gadomski** [comment]
AppVeyor doesn't make a restart easy. I usually amend my last commit `git checkout vs2013-fixes && git commit --amend`, save that commit w/ the same message, then force push that amended commit `git push -f origin vs2013-fixes` (if **origin** is `https://github.com/cffk/vs2013-fixes`).

**Comment by cffk** [comment]
Well, here's the failure that AppVeyor reports.  I don't know where to start to figure out what's causing this!

MSBUILD : error MSB4166: Child node
"Microsoft.Build.Exceptions.InternalLoggerException:
The build stopped unexpectedly because of an unexpected logger failure.
---> System.Net.WebException: The underlying connection was closed:
A connection that was expected to be kept alive was closed by the server.
---> System.IO.IOException: Unable to read data from the transport connection:
An existing connection was forcibly closed by the remote host.
 ---> System.Net.Sockets.SocketException:
An existing connection was forcibly closed by the remote host

**Comment by hobu** [comment]
> "Microsoft.Build.Exceptions.InternalLoggerException:

I did some googling on that, and it appears to be on AppVeyor's end. Something to do with where (azure or google apps) they are hosting the vm that is doing the building of the project. I was hoping to make a support ticket with them in the future to attempt to get to the bottom of it.

File: vendor/arbiter/arbiter.cpp [hunk 1]
```diff
@@ -1419,7 +1419,7 @@ namespace
                         std::isspace(c) &&
                         (out.empty() || std::isspace(out.back())))
                     {
-                        return out;
+                        return out + "";
```
Comment (hobu) [inline vendor/arbiter/arbiter.cpp]: Was this a real change or just lint? I'm not sure I understand...

**Comment by hobu** [comment]
Moved activity to #1442

**Pull request closed without merging**